jdaviz/core/template_mixin.py

```diff
@@ -1901,6 +1907,151 @@ def __init__(self, *args, **kwargs):
+            None, will loop through all references.
+        """
+        # NOTE: is_not_composite is assumed in _get_mark_coords
+        super().__init__(plugin,
+                         items=items,
+                         selected=selected,
+                         multiselect=multiselect,
+                         filters=['is_spatial', 'is_not_composite', 'is_not_annulus'],
+                         dataset=dataset,
+                         viewers=viewers,
+                         default_text=None)
+
+        self.add_traitlets(radius_factor=radius_factor)
+
+        self.add_observe('is_active', self._plugin_active_changed)
+        self.add_observe(selected, self._update_mark_coords)
+        self.add_observe(radius_factor, self._update_mark_coords)
+        # add marks to any new viewers
+        self.hub.subscribe(self, ViewerAddedMessage, handler=self._update_mark_coords)
+        # update coordinates when link type or reference data is _mode_changed
+        self.hub.subscribe(self, LinkUpdatedMessage, handler=self._update_mark_coords)
+
+    def _plugin_active_changed(self, *args):
+        for mark in self.marks:
+            mark.visible = self.plugin.is_active
+
+    @property
+    def image_viewers(self):
+        return [viewer for viewer in self.app._viewer_store.values()
+                if isinstance(viewer, BqplotImageView)]
+
+    @property
+    def marks(self):
+        # NOTE: this will require additional logic if we want to support multiple independent
+        # ApertureSubsetSelect instances per-plugin
+        all_aperture_marks = []
+        for viewer in self.image_viewers:
+            # search for existing mark
+            matches = [mark for mark in viewer.figure.marks
+                       if isinstance(mark, ApertureMark)]
+            if len(matches):
+                all_aperture_marks += matches
+                continue
+
+            x_coords, y_coords = self._get_mark_coords(viewer)
+
+            mark = ApertureMark(
+                viewer,
+                x=x_coords,
+                y=y_coords,
+                colors=['#c75109'],
+                fill_opacities=[0.0],
+                visible=self.plugin.is_active)
+            all_aperture_marks.append(mark)
+            viewer.figure.marks = viewer.figure.marks + [mark]
+        return all_aperture_marks
+
+    def _get_mark_coords(self, viewer):
+        if not len(self.selected) or not len(self.dataset.selected):
+            return [], []
+
+        subset_dicts = self.selected_obj if self.multiselect else [self.selected_obj]
+
+        x_coords, y_coords = np.array([]), np.array([])
+        for subset_dict in subset_dicts:
+            # make a copy so changing the radius doesn't change the cached version in memory
+            # NOTE: this [0] assumes a single subcomponent, which should be safe given the
+            # is_not_composite filter applied
+            roi = subset_dict[0]['subset_state'].roi.copy()
```

Comment:
What about composite subset? Or do those never get so far in the logic route?

Reply:
see the in-line comment directly above 😉

Reply:
(although I have seen cases where editing a subset after it has already been added to the dropdown and selected won't later exclude it... I am working to track that down and will file a bug ticket to fix that independent of this if needed)

Reply:
Okay, assuming that the "cone extraction" (which is the main motivation for this feature) will never work on composite subset, then we good. Thanks!